### ISSUE TITLE
NodeCG with login Enabled results in Strict MIME type checking is enforced for module scripts

### ISSUE THREAD

**Issue by ElyFura** [nodecg/nodecg#466]
Tried to get NodeCG running on my Server everything works fine so far except for OBS got Problems with the Login.

Tested on: NodeCG 0.8 - 1.3.1

```
obs-browser: Failed to load module script: The server responded with a non-JavaScript MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec
```
Source http://localhost:9090/login/:0

Disabling the Login function in NodeCG resolves it but i need it for safety

**Comment by Lange** [comment]
I admit this is not something I've come across before. Can you provide any more details about your setup and the symptoms you are experiencing?

**Comment by ElyFura** [comment]
It´s a Dedicated Server with Debian 9.6

Nodejs: 10.15.1 LTS
NodeCG: 0.8 - 1.3.1

With twitch login enabled on nodecg, also enabled dev mode to see if there are any messages or errors but nothing, tried 4 different browsers without problems only OBS gave me an error.

Tried it with the Cusom OBS from GDQ and the latest avaiable OBS version

OBS Log:
`15:52:09.562: obs-browser: Failed to load module script: The server responded with a non-JavaScript MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec. (source: http://node.xbloodxmoonx.de:9090/login/:0)`

`15:52:25.570: obs-browser: Refused to apply style from 'http://node.xbloodxmoonx.de:9090/bower_components/bootstrap/dist/css/bootstrap.min.css' because its MIME type ('text/html') is not a supported stylesheet MIME type, and strict MIME checking is enabled. (source: http://node.xbloodxmoonx.de:9090/authError?code=token_invalidated&message=This%20token%20has%20been%20invalidated&viewUrl=http://node.xbloodxmoonx.de:9090/bundles/nodecg-dokomi/graphics/index.html:0)`

`15:52:25.599: obs-browser: Refused to apply style from 'http://node.xbloodxmoonx.de:9090/bower_components/bsw-yeti/css/bootstrap.min.css' because its MIME type ('text/html') is not a supported stylesheet MIME type, and strict MIME checking is enabled. (source: http://node.xbloodxmoonx.de:9090/authError?code=token_invalidated&message=This%20token%20has%20been%20invalidated&viewUrl=http://node.xbloodxmoonx.de:9090/bundles/nodecg-dokomi/graphics/index.html:0)`


After disabling the login in nodecg everything works fine

With login enabled only the background of the graphic shows up, just the css nothing else

**Comment by Hoishin** [comment]
I noticed this when I used something `<link rel="stylesheet" href="style.css">` inside nodecg. Probably have to explicitly set mime type for specific file types.

**Comment by Lange** [comment]
I've looked into this a bit more but am still confused as to what is causing it. We use `.sendFile`, which is supposed to automatically infer and set the MIME type.

**Comment by Lange** [comment]
Huh. It looks like this is sometimes happening when the result of the request is a 404, but I don't know why that would make the browser complain about a MIME type instead of just saying there was a 404.

**Comment by Lange** [comment]
I... guess we have to still set the MIME type appropriately, even if we're responding with a 404? This is not something I was aware of.

**Comment by Lange** [comment]
Should be fixed in #546
